lib/chef/provider/execute.rb

```diff
@@ -80,7 +80,9 @@ def opts
         opts[:umask]       = umask if umask
         opts[:log_level]   = :info
         opts[:log_tag]     = new_resource.to_s
-        if STDOUT.tty? && !Chef::Config[:daemon] && Chef::Log.info? && !sensitive?
+        if (Chef::Config[:always_stream_execute] || run_context.events.formatter?) && !sensitive?
+          opts[:live_stream] = Chef::EventDispatch::EventsOutputStream.new(run_context.events, :name => :execute)
+        elsif STDOUT.tty? && !Chef::Config[:daemon] && Chef::Log.info? && !sensitive?
           opts[:live_stream] = STDOUT
         end
```

Comment:
Yeah, I get that I just I just see `!sensitive?` hanging off both of those clauses, so it could be this:

```ruby
unless sensitive?
  if ...
  elseif ...
  end
end
```

And that has the nice property that you can glance at it and see that if sensitive is set you skip everything, which makes it easy to keep the expression correct in that case, which is an important case to those users.   And then you've got the two cases which are messy but which stand alone but conceptually both of those are single cases.  You could even bust it out into helpers where it'd just become:

```ruby
unless sensitive?
  if should_stream_to_formatter?
  elsif should_stream_to_stdout?
  end
end
```

But it is just kinda inherently messy.